lib/srv/authhandlers.go

```diff
@@ -525,6 +545,10 @@ func (h *AuthHandlers) IsHostAuthority(cert ssh.PublicKey, address string) bool
 // client) to see if this certificate can be allowed to login as user:login
 // pair to requested server.
 func (h *AuthHandlers) canLoginWithoutRBAC(cert *ssh.Certificate, clusterName string, teleportUser, osUser string) error {
+	if h.c.AuthorizeWithoutRBAC != nil {
+		return h.c.AuthorizeWithoutRBAC(cert, clusterName, teleportUser, osUser)
+	}
+
```

Comment:
Can we change the abstraction here? This looks hacky in my opinion. You call `canLoginWithoutRBAC` but when `AuthorizeWithoutRBAC` is set RBAC logic won't be called at all only for testing purposes.
I think it would be cleaner to extract `canLoginWithoutRBAC` and `canLoginWithRBAC` as interfaces and provide a different implementation in the production code and tests. In case you forgot to set the code will panic which I think it's better then calling mock by mistake.

Before we go that path, what stops us from running RBAC check in tests? Sounds like something without many side effects.

Reply:
We can run an RBAC check in the tests I added, but I don't think it would be necessary. There are already other tests that confirm that RBAC works correctly, so my thought was instead of needlessly complicating the test by creating users, roles, etc so RBAC will work, just test the a RBAC check happens or doesn't happen.

Reply:
Done